src/components/Pickers/Lookup/Lookup.js

```diff
@@ -58,13 +62,70 @@ class Lookup extends React.PureComponent {
 		}
 	};
 
+	static getDerivedStateFromProps(props, state) {
+		const { search, optionsCache, customOptions } = state;
+		const currentOptions = optionsCache[search] || initialCache;
+		let currentControlOptions = state.controlOptions;
+		let currentValues = optionsCache[search] ? optionsCache[search].values : [];
+		const optionItem = props.value ? Lookup.mapValue(props, state) : null;
+		const currentOptionExists = optionItem && props.value && currentOptions && currentOptions.options.length 
+			? find(currentOptions.options, option => isEqual(option, optionItem))
+			: false;			
+		let availableOptions = currentOptions ? currentOptions.options.concat(customOptions) : [];
+		const isNeedChangeState = !currentOptionExists && optionItem && state.optionsCache[state.search] && !props.isMulti;
+		if (isNeedChangeState) {
+			availableOptions.push(optionItem);
+			currentValues.push(props.value);
+		}
+		Lookup.mapExistingOptionsWithNewValues(availableOptions, currentControlOptions, optionItem);
```

Comment:
I think this expression can be executed inside `if (isNeedChangeState)` block. If the state is not going to be updated this mapping makes no sense

Reply:
No, this method is the only one that updates controlOptions array from loaded items also and controlOptions array contains lookup options.  But method name is not suitable for its functionality, I'll rename it